src/handler.ts

```diff
@@ -1,94 +1,807 @@
+        super: { include: { block_parents: true } }
+      }
+    });
+
+    return respond(block, metagraphBlockResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const currencyTransactions = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { identifier: metagraph_id } = event.pathParameters || {};
+
+    const where = { where: { metagraph_blocks: { metagraph_id } } };
+
+    return metagraphTransactionsQuery(where, event);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const currencyTransaction = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { identifier: metagraph_id, hash } = event.pathParameters || {};
+
+    const transaction = await prisma.metagraph_transactions.findUnique({
+      where: {  metagraph_id_hash: {
+                  metagraph_id: metagraph_id!,
+                  hash: hash!
+                }
+         },
+      include: {
+        metagraph_blocks: {
+          include: {
+            metagraph_snapshots: { select: { hash: true, ordinal: true } }
+          }
+        }
+      }
+    });
+
+    return respond(transaction, metagraphTransactionResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const currencyTransactionsByAddress = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { identifier: metagraph_id, address } = event.pathParameters || {};
+
+    const where = {
+      where: { metagraph_id, 
+        OR: [{ source_addr: address }, { destination_addr: address }]
+      }
+    };
+
+    return metagraphTransactionsQuery(where, event);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const currencyTransactionsBySource = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { identifier: metagraph_id, address } = event.pathParameters || {};
+    
+    const where = { where: { metagraph_id, source_addr: address } };
+
+    return metagraphTransactionsQuery(where, event);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const currencyTransactionsByDestination = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { identifier: metagraph_id, address } = event.pathParameters || {};
+    
+    const where = { where: { metagraph_id, destination_addr: address } };
+
+    return metagraphTransactionsQuery(where, event);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const currencyBalanceByAddress = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { identifier: metagraph_id, address, ordinal } = event.pathParameters || {};
+
+    const ordinalNbr = toNumber(ordinal);
+    const ordinalCondition = (isFinite(ordinalNbr)? {metagraph_snapshot_ordinal: ordinalNbr}: {})
```

Comment:
I think this needs to be `WHERE metagraph_snapshot_ordinal < :ordinal` - this looks like = to me.